### PR TITLE
Check for maintenance

### PR DESCRIPTION
**Pull request by achtenhagen**
This PR migrates the `checkForMaintenance` function to Swift, updates localizations and introduces the wallet options enum.

### CODE REVIEW

File: Blockchain/Alerts/AlertViewPresenter.swift [hunk 1]
```diff
@@ -53,6 +53,10 @@ import Foundation
         }
     }
 
+    func showMaintenanceAlert(withTitle title: String, _ message: String) {
```
Comment (bchrisarriola) [inline Blockchain/Alerts/AlertViewPresenter.swift]: This method might not actually be necessary. Can call `standardNotify(message,title)` directly

File: Blockchain/Authentication/AuthenticationCoordinator+TouchID.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    // TODO: migrate to the responsible controller that prompts for authentication
+    func handleBiometricAuthenticationError(with error: AuthenticationError) {
+        if let description = error.description {
+            let alert = UIAlertController(title: LocalizationConstants.Errors.error, message: description, preferredStyle: .alert)
+            let action = UIAlertAction(title: LocalizationConstants.ok, style: .default, handler: nil)
+            alert.addAction(action)
+            DispatchQueue.main.async {
+                UIApplication.shared.keyWindow?.rootViewController?.present(alert, animated: true, completion: nil)
+            }
```
Comment (bchrisarriola) [inline Blockchain/Authentication/AuthenticationCoordinator+TouchID.swift]: Can also just call `AlertViewPresenter.shared.standardNotify` directly

File: Blockchain/Authentication/AuthenticationCoordinator.swift [hunk 3]
```diff
@@ -133,11 +138,21 @@ import Foundation
...
+//                [self authenticateWithTouchID];
+//            }
+//            #endif
+            authenticateWithBiometrics()
+            //
         } else {
-            // TODO: check for maintenance
-//            [self checkForMaintenance];
+            NetworkManager.shared.checkForMaintenance(withCompletion: { response in
+                if let message = response {
```
Comment (bchrisarriola) [inline Blockchain/Authentication/AuthenticationCoordinator.swift]: Nit: use `guard` instead?

File: Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift [hunk 4]
```diff
@@ -27,8 +27,8 @@ extension AuthenticationCoordinator: PEPinEntryControllerDelegate {
 
         showVerifyingBusyView(withTimeout: 30)
 
-        let pinKey = BlockchainSettings.App.shared.pinKey
-        let pinString = pin.toString
+        // let pinKey = BlockchainSettings.App.shared.pinKey
+        // let pinString = pin.toString
```
Comment (bchrisarriola) [inline Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift]: Remove?

File: Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift [hunk 5]
```diff
@@ -37,12 +37,20 @@ extension AuthenticationCoordinator: PEPinEntryControllerDelegate {
...
-        //            [self checkForMaintenanceWithPinKey:pinKey pin:pin];
-        //        });
+        guard let pinKey = BlockchainSettings.App.shared.pinKey else {
+            return
+        }
 
+        // Check for maintenance before allowing pin entry
+        NetworkManager.shared.checkForMaintenance(withCompletion: { response in
+            LoadingViewPresenter.shared.hideBusyView()
+            guard let message = response else {
```
Comment (bchrisarriola) [inline Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift]: A bit hard to follow the flow in this block. How about flipping the `guard` to check that the response is nil, otherwise, display the error? I think `guard` is best used to check success cases and not the other way around.

So:

```
guard response == nil else {
    print("Error checking for maintenance in wallet options: %@", message)
    self.pinEntryViewController?.reset()
    AlertViewPresenter.shared.showMaintenanceAlert(withTitle: LocalizationConstants.Errors.error, message)
    return
}
self.walletManager.wallet.apiGetPINValue(pinKey, pin: pin.toString)
```

File: Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift [hunk 6]
```diff
@@ -37,12 +37,20 @@ extension AuthenticationCoordinator: PEPinEntryControllerDelegate {
...
-
-        //        dispatch_async(dispatch_get_main_queue(), ^{
-        //            [self checkForMaintenanceWithPinKey:pinKey pin:pin];
-        //        });
+        guard let pinKey = BlockchainSettings.App.shared.pinKey else {
+            return
+        }
 
+        // Check for maintenance before allowing pin entry
+        NetworkManager.shared.checkForMaintenance(withCompletion: { response in
```
Comment (bchrisarriola) [inline Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift]: Retain cycle here—add `[weak self]`

i.e.
```
    NetworkManager.shared.checkForMaintenance(withCompletion: { [weak self] response in
        guard let strongSelf = self else { return }
        // ...do stuff
    })
```

File: Blockchain/Coordinators/OnboardingCoordinator.swift [hunk 7]
```diff
@@ -12,21 +12,24 @@ import Foundation
...
 
-    private init() {
-    }
+    private init() {}
 
     // MARK: Public Methods
 
     func start() {
-        showWelcomeScreen()
+        NetworkManager.shared.checkForMaintenance(withCompletion: { response in
```
Comment (bchrisarriola) [inline Blockchain/Coordinators/OnboardingCoordinator.swift]: Retain cycle—add `[weak self]`

File: Blockchain/Network/NetworkManager.swift [hunk 8]
```diff
@@ -53,6 +53,36 @@ final class NetworkManager: NSObject, URLSessionDelegate {
...
     }
 
+    func checkForMaintenance(withCompletion handler: @escaping (_ response: String?) -> Void) {
+        guard
+            let walletOptionsUrl = BlockchainAPI.shared.walletOptionsUrl,
+            let url = URL(string: walletOptionsUrl) else {
+                fatalError("Failed to get wallet options url from Bundle.")
+        }
+        NetworkManager.shared.session.sessionDescription = url.host
+        let task = NetworkManager.shared.session.dataTask(with: url) { data, _, error in
```
Comment (bchrisarriola) [inline Blockchain/Network/NetworkManager.swift]: `[weak self]`

File: Blockchain/Network/NetworkManager.swift [hunk 9]
```diff
@@ -53,6 +53,36 @@ final class NetworkManager: NSObject, URLSessionDelegate {
...
+            if error != nil {
```
Comment (bchrisarriola) [inline Blockchain/Network/NetworkManager.swift]: `guard` instead?

File: Blockchain/Wallet/WalletOptions.swift [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+//  WalletOptions.swift
+//  Blockchain
+//
+//  Created by Maurice A. on 5/1/18.
+//  Copyright © 2018 Blockchain Luxembourg S.A. All rights reserved.
+//
+
+import Foundation
+
+enum WalletOptions: String {
```
Comment (bchrisarriola) [inline Blockchain/Wallet/WalletOptions.swift]: Should this actually be a `struct`?

File: Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift (hunk 4 above)
Comment (achtenhagen) [reply Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift]: 👍

File: Blockchain/Alerts/AlertViewPresenter.swift (hunk 1 above)
Comment (achtenhagen) [reply Blockchain/Alerts/AlertViewPresenter.swift]: 👍

File: Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift (hunk 5 above)
Comment (achtenhagen) [reply Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift]: 👍

File: Blockchain/Wallet/WalletOptions.swift (hunk 10 above)
Comment (achtenhagen) [reply Blockchain/Wallet/WalletOptions.swift]: I made it an enum because it isn't yet clear how we are going to use this, but making it a struct is probably what will end up happening.

File: Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift (hunk 6 above)
Comment (achtenhagen) [reply Blockchain/Authentication/Models/AuthenticationCoordinator+Pin.swift]: In the case of Authentication, I think `[unowned self]` would work. Agree?

File: Blockchain/Wallet/WalletOptions.swift (hunk 10 above)
Comment (achtenhagen) [reply Blockchain/Wallet/WalletOptions.swift]: Converted to `struct`.

File: Blockchain/Network/NetworkManager.swift (hunk 9 above)
Comment (achtenhagen) [reply Blockchain/Network/NetworkManager.swift]: 👍

File: Blockchain/Authentication/AuthenticationCoordinator.swift (hunk 3 above)
Comment (achtenhagen) [reply Blockchain/Authentication/AuthenticationCoordinator.swift]: 👍

**Comment by achtenhagen** [comment]
Thanks for your feedback! Please let me know what you think about my comments and merge when ready!

**Comment by bchrisarriola** [comment]
LGTM. Good to merge into `swift`, waiting to merge #133 first.
